samples/bidirectional-chat/csharp/Function.cs

```diff
@@ -0,0 +1,124 @@
+        }
+
+        [FunctionAuthorize]
+        [FunctionName(nameof(Broadcast))]
+        public async Task Broadcast([SignalRTrigger]InvocationContext invocationContext, [SignalRParameter]string message)
+        {
+            await Clients.All.SendAsync(NewMessageTarget, new NewMessage(invocationContext, message));
+        }
+
+        [FunctionName(nameof(SendToGroup))]
+        public async Task SendToGroup([SignalRTrigger]InvocationContext invocationContext,
+            [SignalRParameter]string groupName,
+            [SignalRParameter]string message)
+        {
+            await Clients.Group(groupName).SendAsync(NewMessageTarget, new NewMessage(invocationContext, message));
+        }
+
+        [FunctionName(nameof(SendToUser))]
+        public async Task SendToUser([SignalRTrigger]InvocationContext invocationContext,
+            [SignalRParameter]string userName,
+            [SignalRParameter]string message)
+        {
+            await Clients.User(userName).SendAsync(NewMessageTarget, new NewMessage(invocationContext, message));
+        }
+
+        [FunctionName(nameof(SendToConnection))]
+        public async Task SendToConnection([SignalRTrigger]InvocationContext invocationContext,
+            [SignalRParameter]string connectionId,
+            [SignalRParameter]string message)
+        {
+            await Clients.Client(connectionId).SendAsync(NewMessageTarget, new NewMessage(invocationContext, message));
+        }
+
+        [FunctionName(nameof(JoinGroup))]
+        public async Task JoinGroup([SignalRTrigger]InvocationContext invocationContext,
+            [SignalRParameter]string connectionId,
+            [SignalRParameter]string groupName)
+        {
+            await Groups.AddToGroupAsync(connectionId, groupName);
+        }
+
+        [FunctionName(nameof(LeaveGroup))]
+        public async Task LeaveGroup([SignalRTrigger]InvocationContext invocationContext,
+            [SignalRParameter]string connectionId,
+            [SignalRParameter]string groupName)
+        {
+            await Groups.RemoveFromGroupAsync(connectionId, groupName);
+        }
+
+        [FunctionName(nameof(JoinUserToGroup))]
+        public async Task JoinUserToGroup([SignalRTrigger]InvocationContext invocationContext,
+            [SignalRParameter]string userName,
+            [SignalRParameter]string groupName)
+        {
+            await UserGroups.AddToGroupAsync(userName, groupName);
+        }
+
+        [FunctionName(nameof(LeaveUserFromGroup))]
+        public async Task LeaveUserFromGroup([SignalRTrigger]InvocationContext invocationContext,
+            [SignalRParameter]string userName,
+            [SignalRParameter]string groupName)
+        {
+            await UserGroups.RemoveFromGroupAsync(userName, groupName);
+        }
+
+        [FunctionName(nameof(Disconnect))]
+        public void Disconnect([SignalRTrigger]InvocationContext invocationContext)
```

Comment:
Can you call these methods explicitly?

Reply:
What's your "explicitly" means?

Reply:
Can these calls come from outside the signalr service?

Reply:
Can the client trigger a call to disconnect and not be disconnected?

Reply:
Yes, as long as the message sent to SignalR trigger can follow our serverless spec.

Reply:
If using SignalR Service, it won't happen. Because all message from client's invocation will be in the `messages` category. And connected/disconnected are `connections` category